test/es6/SirenActionTest.js

```diff
@@ -1,185 +1,146 @@
+			fetchMock.deleteOnce('http://api.x.io/orders/42/items/5', {});
+			await performSirenActions('token', [AFitemDelete1]);
+			expect(fetchMock.called()).to.be.true;
+		});
+
+		it('sends 1 request when working with 1 action with no (i.e. empty array) fields', async() => {
+			fetchMock.deleteOnce('http://api.x.io/orders/42/items/5', {});
+			AFitemDelete1.fields = [];
+			await performSirenActions('token', [AFitemDelete1]);
+			expect(fetchMock.called()).to.be.true;
+		});
+
+		it('sends 1 request when working with 1 action with fields', async() => {
+			fetchMock.postOnce('http://api.x.io/orders/42/items', {});
+
+			await performSirenActions('token', [AFitemsPost1]);
+
+			const form = await getFormData(fetchMock.lastCall().request);
+			if (!form.notSupported) {
+				expect(form.get('orderNumber')).to.equal('42');
+				expect(form.get('productCode')).to.equal('10000');
+			}
+			expect(fetchMock.called()).to.be.true;
+		});
+
+		it('sends 2 requests when working with 2 actions with different endpoints', async() => {
+			fetchMock.postOnce('http://api.x.io/orders/42/items', {});
+			fetchMock.patchOnce('http://api.x.io/orders/42/items/5', {});
+
+			await performSirenActions('token', [AFitemsPost1, AFitemPatch1]);
+
+			let form = await getFormData(fetchMock.calls()[0].request);
+			if (!form.notSupported) {
+				expect(form.get('orderNumber')).to.equal('42');
+				expect(form.get('productCode')).to.equal('10000');
+			}
+
+			form = await getFormData(fetchMock.calls()[1].request);
+			if (!form.notSupported) {
+				expect(form.get('orderNumber')).to.equal('50');
+				expect(form.get('productCode')).to.equal('20000');
+			}
+
+			expect(fetchMock.done()).to.be.true;
+		});
+
+		it('sends 2 requests when working with 2 actions with the same endpoint but different methods', async() => {
+			fetchMock.patchOnce('http://api.x.io/orders/42/items/5', {});
+			fetchMock.deleteOnce('http://api.x.io/orders/42/items/5', {});
+
+			await performSirenActions('token', [AFitemPatch1, AFitemDelete1]);
+
+			const form = await getFormData(fetchMock.calls()[0].request);
+			if (!form.notSupported) {
+				expect(form.get('orderNumber')).to.equal('50');
+				expect(form.get('productCode')).to.equal('20000');
+			}
+
+			expect(fetchMock.done()).to.be.true;
+		});
+
+		it('sends 1 request when working with 2 actions with the same endpoint and the same method', async() => {
+			fetchMock.patchOnce('http://api.x.io/orders/42/items/5', {});
+
+			await performSirenActions('token', [AFitemPatch1, AFitemPatch2]);
+
+			const form = await getFormData(fetchMock.lastCall().request);
+			if (!form.notSupported) {
+				expect(form.get('orderNumber')).to.equal('50');
+				expect(form.get('productCode')).to.equal('20000');
+				expect(form.get('visibleToCustomer')).to.equal('true');
+				expect(form.get('quantity')).to.equal('10');
+			}
+
+			expect(fetchMock.called()).to.be.true;
+		});
+	});
+
+	describe('enqueue action', () => {

```

Comment:
Did you mean to keep the old tests here or was this left by mistake?

Reply:
oops lol